featuretools/entityset/entityset.py

```diff
@@ -1008,66 +1006,79 @@ def add_last_time_indexes(self, updated_dataframes=None):
-    #                         skip = True
-    #                         break
-    #                 if skip:
-    #                     continue
-
-    #                 counts = entity.df[variable.id].value_counts()
-
-    #                 # find how many of each unique value there are; sort by count,
-    #                 # and add interesting values to each variable
-    #                 total_count = np.sum(counts)
-    #                 counts[:] = counts.sort_values()[::-1]
-    #                 for i in range(min(max_values, len(counts.index))):
-    #                     idx = counts.index[i]
-
-    #                     if len(counts.index) < 25:
-    #                         if verbose:
-    #                             msg = "Variable {}: Marking {} as an "
-    #                             msg += "interesting value"
-    #                             logger.info(msg.format(variable.id, idx))
-    #                         variable.interesting_values = variable.interesting_values.append(pd.Series([idx]))
-    #                     else:
-    #                         fraction = counts[idx] / total_count
-    #                         if fraction > 0.05 and fraction < 0.95:
-    #                             if verbose:
-    #                                 msg = "Variable {}: Marking {} as an "
-    #                                 msg += "interesting value"
-    #                                 logger.info(msg.format(variable.id, idx))
-    #                             variable.interesting_values = variable.interesting_values.append(pd.Series([idx]))
-    #                         else:
-    #                             break
+        """
+        if dataframe_id is None and values is not None:
+            raise ValueError("dataframe_id must be specified if values are provided")
 
-    #     self.reset_data_description()
+        if dataframe_id is not None and values is not None:
+            for column, vals in values.items():
+                self[dataframe_id].ww.columns[column].metadata['interesting_values'] = vals
+            return
+
+        if dataframe_id:
+            dataframes = [self[dataframe_id]]
+        else:
+            dataframes = self.dataframes
+        for df in dataframes:
+            for column in df.columns:
+                # some heuristics to find basic 'where'-able variables
+                if df.ww.columns[column].is_categorical:
+                    # TODO - consider removing this constraints
+                    # don't add interesting values for entities in relationships
+                    skip = False
+                    for r in self.relationships:
+                        if column in [r.child_column.name, r.parent_column.name]:
+                            skip = True
+                            break
+                    if skip:
+                        continue
+
+                    counts = df[column].value_counts()
```

Comment:
could we make use of woodwork's value_counts here? It already calculates value counts only for columns that are `is_categorical`.

I guess to use it, we'd end up calling it for the dataframe as a whole and then having to go through column by column and get the interesting values after the fact, so it'd be a structural change here, but thought I'd bring it up!

Reply:
Interesting idea! Might allow us to eliminate some looping here and simplify this code a bit. I'll create an issue to look into that further.

Reply:
https://github.com/alteryx/featuretools/issues/1426